src/BayesFilters/include/BayesFilters/UKFPrediction.h

```diff
@@ -0,0 +1,55 @@
+#ifndef UKFPREDICTION_H
+#define UKFPREDICTION_H
+
+#include <BayesFilters/AdditiveStateModel.h>
+#include <BayesFilters/ExogenousModel.h>
+#include <BayesFilters/GaussianMixture.h>
+#include <BayesFilters/GaussianPrediction.h>
+#include <BayesFilters/sigma_point.h>
+#include <BayesFilters/StateModel.h>
+
+#include <memory>
+
+namespace bfl {
+    class UKFPrediction;
+}
+
+
+enum class UKFPredictionType { Generic, Additive };
```

Comment:
This `enum class` seems not to be usable by the user since `UKFPredictionType` member variable is set by the constructor depending on the `StateModel` type. Consequently, it should be moved inside the class in the `protected` scope.